src/components/Composer/index.tsx

```diff
@@ -1,134 +1,33 @@
-    onPasteFile: PropTypes.func,
-
-    /** A ref to forward to the text input */
-    forwardedRef: PropTypes.func,
-
-    /** General styles to apply to the text input */
-    // eslint-disable-next-line react/forbid-prop-types
-    style: PropTypes.any,
-
-    /** If the input should clear, it actually gets intercepted instead of .clear() */
-    shouldClear: PropTypes.bool,
-
-    /** When the input has cleared whoever owns this input should know about it */
-    onClear: PropTypes.func,
-
-    /** Whether or not this TextInput is disabled. */
-    isDisabled: PropTypes.bool,
-
-    /** Set focus to this component the first time it renders.
-  Override this in case you need to set focus on one field out of many, or when you want to disable autoFocus */
-    autoFocus: PropTypes.bool,
-
-    /** Update selection position on change */
-    onSelectionChange: PropTypes.func,
-
-    /** Selection Object */
-    selection: PropTypes.shape({
-        start: PropTypes.number,
-        end: PropTypes.number,
-    }),
-
-    /** Whether the full composer can be opened */
-    isFullComposerAvailable: PropTypes.bool,
-
-    /** Allow the full composer to be opened */
-    setIsFullComposerAvailable: PropTypes.func,
-
-    /** Should we calculate the caret position */
-    shouldCalculateCaretPosition: PropTypes.bool,
-
-    /** Function to check whether composer is covered up or not */
-    checkComposerVisibility: PropTypes.func,
-
-    /** Whether this is the report action compose */
-    isReportActionCompose: PropTypes.bool,
-
-    /** Whether the sull composer is open */
-    isComposerFullSize: PropTypes.bool,
-
-    /** Should make the input only scroll inside the element avoid scroll out to parent */
-    shouldContainScroll: PropTypes.bool,
-
-    ...withLocalizePropTypes,
-};
-
-const defaultProps = {
-    defaultValue: undefined,
-    value: undefined,
-    numberOfLines: 0,
-    onNumberOfLinesChange: () => {},
-    maxLines: -1,
-    onPasteFile: () => {},
-    shouldClear: false,
-    onClear: () => {},
-    style: null,
-    isDisabled: false,
-    autoFocus: false,
-    forwardedRef: null,
-    onSelectionChange: () => {},
-    selection: {
-        start: 0,
-        end: 0,
-    },
-    isFullComposerAvailable: false,
-    setIsFullComposerAvailable: () => {},
-    shouldCalculateCaretPosition: false,
-    checkComposerVisibility: () => false,
-    isReportActionCompose: false,
-    isComposerFullSize: false,
-    shouldContainScroll: false,
-};
+import {ComposerProps} from './types';
 
 /**
  * Retrieves the characters from the specified cursor position up to the next space or new line.
  *
- * @param {string} str - The input string.
- * @param {number} cursorPos - The position of the cursor within the input string.
- * @returns {string} - The substring from the cursor position up to the next space or new line.
+ * @param str - The input string.
+ * @param cursorPos - The position of the cursor within the input string.
```

Comment:
NAB: `inputString` and `cursorPosition` instead of abbreviated variable names (same with `subString` on line 32 if you do end up changing this)

Reply:
Great idea, always an improvement 😄